detect_secrets/plugins/jwt.py

```diff
@@ -19,6 +22,7 @@ class JwtTokenDetector(RegexBasedDetector):
     def analyze_string(self, string: str) -> Generator[str, None, None]:
         yield from filter(
             self.is_formally_valid,
+            self.is_time_valid,
```

Comment:
Uh...does this work?

Based on the [python documentation](https://docs.python.org/3/library/functions.html#filter), `filter` takes in a function, and an iterable. Based on this, I'm surprised that three arguments would actually pass tests...